tests/create-testing-library-rule.test.ts

```diff
@@ -229,32 +264,69 @@ ruleTester.run(RULE_NAME, rule, {
       findByRole('button')
     `,
     },
+
+    // Test Cases for async utils
     {
       settings: {
-        'testing-library/module': 'test-utils',
+        'testing-library/utils-module': 'test-utils',
       },
       code: `
-        import * as tl from 'test-utils'
-        const obj = { tl }
-        obj.tl.waitFor(() => {})
+        import { waitFor } from 'some-other-library';
+        test(
+          'aggressive reporting disabled - util waitFor not related to testing library is valid',
+          () => { waitFor() }
+        );
       `,
     },
     {
-      settings: { 'testing-library/module': 'test-utils' },
+      filename: 'file.not.matching.js',
       code: `
-      // case: aggressive render enabled, but module disabled - not coming from TL
-      import { render } from 'somewhere-else'
+      // case: waitFor util found, but file name not matching
+      import { waitFor } from '@testing-library/react'
       
-      const utils = render()
+      waitFor()
       `,
     },
+
+    // Test Cases for all settings mixed
     {
-      filename: 'file.not.matching.js',
+      settings: {
+        'testing-library/utils-module': 'test-utils',
+        'testing-library/filename-pattern': 'testing-library\\.js',
+      },
       code: `
-      // case: aggressive render and module enabled, but file name not matching
+      // case: matching custom settings partially - module but not filename
+      import { render } from 'test-utils'
+      import { somethingElse } from 'another-module'
+      const foo = require('bar')
+      
+      const utils = render();
+      `,
+    },
+    {
+      settings: {
+        'testing-library/utils-module': 'test-utils',
+        'testing-library/filename-pattern': 'testing-library\\.js',
+      },
+      filename: 'MyComponent.testing-library.js',
+      code: `
+      // case: matching custom settings partially - filename but not module
+      import { render } from 'other-utils'
+      import { somethingElse } from 'another-module'
+      const foo = require('bar')
+      
+      const utils = render();
+      `,
+    },
+    {
+      settings: { 'testing-library/utils-module': 'test-utils' },
+      code: `
+      // case: aggressive module disabled and render coming from non-related module
+      import * as somethingElse from '@somewhere/else'
       import { render } from '@testing-library/react'
       
-      const utils = render()
+      // somethingElse.render is not coming from any module related to TL
+      const utils = somethingElse.render()
```

Comment:
This new test has detected few issues in our implementation! I'm glad I added it at the beginning of this tweaks branch. It will make sure from now on that we check from where are imported those methods in member expressions properly.